exec/supervisor/plugin.go

```diff
@@ -97,11 +100,22 @@ func (p *Plugin) Init() error {
 // Close local resources
 func (p *Plugin) Close() error {
 	for _, program := range p.programs {
+		if program.process.IsAlive() {
+			if _, err := program.process.StopAndWait(); err != nil {
+				p.Log.Errorf("failed to stop program %s: %v", program.process.GetName(), err)
+			}
+		}
 		if err := program.svLogger.Close(); err != nil {
 			p.Log.Errorf("failed to close logger: %v", err)
 		}
+		// terminate watcher for given supervisor process
+		close(program.doneChan)
 	}
+
+	// close hook watcher when all programs are terminated
+	p.wg.Wait()
 	close(p.hookChan)
```

Comment:
This is improper handling of channel as I've explained in other PR: https://github.com/ligato/vpp-agent/pull/1318#discussion_r288880877

The `Close` method is called from different goroutine than the goroutine that handles `watch` method and sends events to the `hookChan`.

Reply:
Indeed, that's the reason for the wait group. There is a `watch` goroutine for every process started and the hook channel is shared between them. The wait group ensures that no more watchers are running, and then it's safe to close the channel even from a different goroutine.

Reply:
But it doesn't really make much sense why you want to close the `hookChan` when immediately after closing it you return from `Close`.. which means that it will not wait for `watchEvents` to finish anyway so why bother?